master/buildbot/test/unit/reporters/test_base.py

```diff
@@ -189,19 +189,19 @@ def generate_throw(*args, **kwargs):
 
     @defer.inlineCallbacks
     def test_reports_sent_in_order_despite_slow_generator(self):
```

Comment:
I think we also need a similar test where there's a call to `_got_event(..., {'buildrequestid': 2})` in between the calls to `_got_event(..., {'buildrequestid': 1})`. Then give result to the deferred for  `_got_event(..., {'buildrequestid': 2})` before giving result to the other deferreds and verify that processing for buildrequestid==2 was not blocked on processing for buildrequestid==1 (just the order of items in call_args_list is enough)